### ISSUE TITLE
Using the dto class and @Body decorator does not work as expected.

### ISSUE THREAD

**Issue by alexandr2110pro** [nestjs/swagger#65]
The code:

**DTOs:**
```TypeScript
import {
  ApiModelProperty,
} from '@nestjs/swagger';

import { UserDto } from './user.dto';

export class CreateUserDto extends UserDto {

  @ApiModelProperty()
  readonly password: string;

  @ApiModelProperty()
  readonly confirmPassword: string;
}
```

```TypeScript
import {
  ApiModelProperty,
  ApiModelPropertyOptional,
} from '@nestjs/swagger';


export class UserDto {

  @ApiModelPropertyOptional()
  readonly firstName?: string;

  @ApiModelPropertyOptional()
  readonly lastName?: string;

  @ApiModelProperty()
  readonly email: string;
}
```

**Controller:**
```TypeScript

@ApiUseTags('users')
@Controller('users')
export class UsersController {

  constructor(private readonly usersService: UsersService) {}

  /**
   * Creates a new User (register)
   *
   * @param createUserDto
   * @return Promise<void>
   */
  @Post()
  @ApiOperation({ title: 'Create a new User' })
  @ApiResponse({
    status: HttpStatus.CREATED,
    description: 'The record has been successfully created.',
  })
  @ApiResponse({
    status: HttpStatus.BAD_REQUEST,
    description: 'confirm password doesn\'t match the password, or the payload is invalid',
  })
  @ApiResponse({
    status: HttpStatus.CONFLICT,
    description: 'Email already in use',
  })
  @ApiResponse({ status: HttpStatus.FORBIDDEN, description: 'Forbidden.' })
  @ApiResponse({ status: HttpStatus.UNAUTHORIZED, description: 'Unauthorized.' })
  async createUser(@Body() createUserDto: CreateUserDto) {
    await this.usersService
      .create(createUserDto)
      .catch(this._handleError);
  }

  / * <...> */

}
```

The result:

![screenshot 2018-03-22 03 05 24](https://user-images.githubusercontent.com/2496784/37743998-f36d0014-2d7d-11e8-9193-ac57208c7a04.png)

---

However, using the following is fixing the Swagger UI:

```TypeScript
  @ApiImplicitBody({
    name: 'role data',
    type: CreateUserDto,
  })
```

**Comment by alexandr2110pro** [comment]
The same is for the `@Query` decorator

**Comment by milescui** [comment]
I also encountered the same problem

**Comment by kamilmysliwiec** [comment]
I'm trying to reproduce this issue. What about this sample: https://github.com/nestjs/nest/tree/master/sample/11-swagger? Do you face a similar problem? What Nest version do you use?

**Comment by nursik** [comment]
I did not have such a problem before changing dependencies in package.json. It sounds weird, but when I tried to revert to previous dependencies, problem was still here. Surprisingly, I was trying to solve issue with pipes, because built-in pipes did not work for NestJs 5.0-rc.2.

Dependencies that I changed were: rxjs 5.5.6 -> 6.0.0. Some kinda strange

**Comment by kamilmysliwiec** [comment]
This issue is not solved for Nest v5.0.0 yet.

**Comment by alexandr2110pro** [comment]
Small update from my team.

It has started working on the same version. 

I suspect that it has something to do with issues in DTO declarations (our mistakes), like using an Interface as a property type in DTO. If I'm not 100% sure, but I think it has started working after we eliminated such declarations by replacing interfaces with other DTO classes.

---

But now we have another issue: `@ApiBearerAuth` doesn't do a thing! 

Will try to find when it happened and why. So far, I have no idea.

**Comment by FreakTheMighty** [comment]
I believe I'm seeing the same issue for a project that I just returned to. 

```
+-- @nestjs/common@4.6.6
+-- @nestjs/core@4.6.6
+-- @nestjs/microservices@4.6.6
+-- @nestjs/swagger@1.3.1
+-- @nestjs/testing@4.6.6
+-- @nestjs/typeorm@1.0.1
+-- @nestjs/websockets@4.6.6
```

I have this controller

```typescript
@Controller('user')
@ApiUseTags('users')
export class UserController {
  constructor(private readonly userService: UserService) {}

  @Post()
  @UsePipes(new PipesModelParser())
  @ApiOperation({ title: 'register new user' })
  async create(@Body() user: User): Promise<User> {
    const newUser = await this.userService.create(user);
    this.userService.createToken(user);
    return newUser;
  }
  ...
```

And the swagger UI shows no body parameter.

**Comment by FreakTheMighty** [comment]
In the swagger console I get

```
root-injects.js:95 TypeError: Cannot read property 'anyOf' of undefined
    at e (fn.js:77)
    at e (fn.js:60)
    at configure-map.js:43
    at t.getSampleSchema (utils.js:608)
    at t.value (response.jsx:113)
    at t.render (root-injects.js:93)
    at u._renderValidatedComponentWithoutOwnerOrContext (ReactCompositeComponent.js:796)
    at u._renderValidatedComponent (ReactCompositeComponent.js:819)
    at u.performInitialMount (ReactCompositeComponent.js:359)
    at u.mountComponent (ReactCompositeComponent.js:255)
t.render @ root-injects.js:95
_renderValidatedComponentWithoutOwnerOrContext @ ReactCompositeComponent.js:796
_renderValidatedComponent @ ReactCompositeComponent.js:819
performInitialMount @ ReactCompositeComponent.js:359
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
mountChildren @ ReactMultiChild.js:234
_createInitialChildren @ ReactDOMComponent.js:701
mountComponent @ ReactDOMComponent.js:520
mountComponent @ ReactReconciler.js:43
mountChildren @ ReactMultiChild.js:234
_createInitialChildren @ ReactDOMComponent.js:701
mountComponent @ ReactDOMComponent.js:520
mountComponent @ ReactReconciler.js:43
mountChildren @ ReactMultiChild.js:234
_createInitialChildren @ ReactDOMComponent.js:701
mountComponent @ ReactDOMComponent.js:520
mountComponent @ ReactReconciler.js:43
mountChildren @ ReactMultiChild.js:234
_createInitialChildren @ ReactDOMComponent.js:701
mountComponent @ ReactDOMComponent.js:520
mountComponent @ ReactReconciler.js:43
performInitialMount @ ReactCompositeComponent.js:368
mountComponent @ ReactCompositeComponent.js:255
mountComponent @ ReactReconciler.js:43
updateChildren @ ReactChildReconciler.js:119
_reconcilerUpdateChildren @ ReactMultiChild.js:209
_updateChildren @ ReactMultiChild.js:308
updateChildren @ ReactMultiChild.js:295
_updateDOMChildren @ ReactDOMComponent.js:944
updateComponent @ ReactDOMComponent.js:758
receiveComponent @ ReactDOMComponent.js:720
receiveComponent @ ReactReconciler.js:122
updateChildren @ ReactChildReconciler.js:107
_reconcilerUpdateChildren @ ReactMultiChild.js:209
_updateChildren @ ReactMultiChild.js:308
updateChildren @ ReactMultiChild.js:295
_updateDOMChildren @ ReactDOMComponent.js:944
updateComponent @ ReactDOMComponent.js:758
receiveComponent @ ReactDOMComponent.js:720
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
updateChildren @ ReactChildReconciler.js:107
_reconcilerUpdateChildren @ ReactMultiChild.js:209
_updateChildren @ ReactMultiChild.js:308
updateChildren @ ReactMultiChild.js:295
_updateDOMChildren @ ReactDOMComponent.js:944
updateComponent @ ReactDOMComponent.js:758
receiveComponent @ ReactDOMComponent.js:720
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
updateChildren @ ReactChildReconciler.js:107
_reconcilerUpdateChildren @ ReactMultiChild.js:209
_updateChildren @ ReactMultiChild.js:308
updateChildren @ ReactMultiChild.js:295
_updateDOMChildren @ ReactDOMComponent.js:944
updateComponent @ ReactDOMComponent.js:758
receiveComponent @ ReactDOMComponent.js:720
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
updateChildren @ ReactChildReconciler.js:107
_reconcilerUpdateChildren @ ReactMultiChild.js:209
_updateChildren @ ReactMultiChild.js:308
updateChildren @ ReactMultiChild.js:295
_updateDOMChildren @ ReactDOMComponent.js:944
updateComponent @ ReactDOMComponent.js:758
receiveComponent @ ReactDOMComponent.js:720
receiveComponent @ ReactReconciler.js:122
updateChildren @ ReactChildReconciler.js:107
_reconcilerUpdateChildren @ ReactMultiChild.js:209
_updateChildren @ ReactMultiChild.js:308
updateChildren @ ReactMultiChild.js:295
_updateDOMChildren @ ReactDOMComponent.js:944
updateComponent @ ReactDOMComponent.js:758
receiveComponent @ ReactDOMComponent.js:720
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
receiveComponent @ ReactCompositeComponent.js:544
receiveComponent @ ReactReconciler.js:122
_updateRenderedComponent @ ReactCompositeComponent.js:751
_performComponentUpdate @ ReactCompositeComponent.js:721
updateComponent @ ReactCompositeComponent.js:642
performUpdateIfNecessary @ ReactCompositeComponent.js:558
performUpdateIfNecessary @ ReactReconciler.js:154
b @ ReactUpdates.js:148
perform @ Transaction.js:141
perform @ Transaction.js:141
perform @ ReactUpdates.js:87
x @ ReactUpdates.js:170
closeAll @ Transaction.js:207
perform @ Transaction.js:154
batchedUpdates @ ReactDefaultBatchingStrategy.js:60
e @ ReactUpdates.js:198
a @ ReactUpdateQueue.js:22
enqueueSetState @ ReactUpdateQueue.js:216
s.setState @ ReactBaseClasses.js:62
o.handleChange @ connect.js:302
g @ createStore.js:173
(anonymous) @ utils.js:124
(anonymous) @ bindActionCreators.js:3
(anonymous) @ actions.js:221
x @ runtime.js:62
(anonymous) @ runtime.js:296
e.(anonymous function) @ runtime.js:114
r @ asyncToGenerator.js:17
(anonymous) @ asyncToGenerator.js:28
```

**Comment by chadjaros** [comment]
We saw this problem begin when we upgraded to Nest 5.0.0

**Comment by FreakTheMighty** [comment]
@chadjaros I'm seeing this problem with Nest 4 as well. Are you able to confirm things still work for your previous versions?

**Comment by kamilmysliwiec** [comment]
```
But now we have another issue: @ApiBearerAuth doesn't do a thing! It worked previously but suddenly it stopped.
```
@alexandr2110pro track this issue here https://github.com/nestjs/swagger/issues/83

The rest should be fixed in `2.0.1` which I just published.

**Comment by hjzheng** [comment]
any updates for this issue, I have the same problem

```
export class JobListDto {
  @ApiModelPropertyOptional()
  user: string[]

  @ApiModelPropertyOptional()
  status: string[]

  @ApiModelPropertyOptional()
  software: string[]

  @ApiModelPropertyOptional()
  dates: number[]
}

@Get('execl')
  async execl(@Query() query: JobListDto, @Res() res: Response) {
    const data = await this.jobService.excelData(query)
    res.setHeader('Content-Type', 'application/vnd.openxmlformats')
    res.setHeader(
      'Content-Disposition',
      `attachment; filename=job_${new Date().toISOString()}.xlsx`,
    )
    res.end(data, 'binary')
  }
```
